addons/web/static/src/core/py_js/py.js

```diff
@@ -33,6 +33,32 @@ export function parseExpr(expr) {
  * @returns {any}
  */
 export function evaluateExpr(expr, context = {}) {
-    const ast = parseExpr(expr);
-    return evaluate(ast, context);
+    let ast;
+    try {
+        ast = parseExpr(expr);
+    } catch (error) {
+        throw new EvalError(`Can not parse python expression: (${expr})\nError: ${error.message}`);
+    }
+    try {
+        return evaluate(ast, context);
+    } catch (error) {
+        throw new EvalError(`Can not evaluate python expression: (${expr})\nError: ${error.message}`);
+    }
+}
+
+/**
+ * Evaluates a python expression
+ *
+ * @param {string} expr
+ * @param {Object} [context]
+ * @returns {any}
+ */
+export function evaluateBooleanExpr(expr, context = {}) {
```

Comment:
please don't forget to add a bunch of tests

Reply:
Basically, all the changes are made for the tests to pass. Everything is covered by the tests. The tests have been changed to no longer use attrs and I have added some combinations. So what kind of tests would you like?

For example the empty list in python is falsy and not in js hence the evaluation with bool(), there are tests for that seen: those of the kind invisible="not many2many_ids"